tests/test_step_methods.py

```diff
@@ -433,6 +438,19 @@ def test_Hsstep():
     assert beta_samples.shape == (50, M)
     np.testing.assert_allclose(beta_samples.mean(0), beta_true, atol=0.3)
 
+
+    with pm.Model():
+        beta = HorseShoe("beta", tau=1, shape=M, testval=pos_beta_true)
+        pm.NegativeBinomial("y", mu =beta.dot(pos_X.T), alpha=10, observed=y_nb)
+        hsstep = HSStep([beta])
+        trace = pm.sample(
+            draws=50, tune=0, step=hsstep, chains=1, return_inferencedata=True
+        )
+
+    beta_samples = trace.posterior["beta"][0].values
+    assert beta_samples.shape == (50, M)
+    np.testing.assert_allclose(beta_samples.mean(0), pos_beta_true, atol=1.5)
+
```

Comment:
These tests would be better off separated into distinct test functions (or parameterized).